File: 151003/Baranouski/RV_Project/src/main/java/by/bsuir/RV_Project/dto/requests/converters/StoryRequestConverter.java
package by.bsuir.RV_Project.dto.requests.converters;

import by.bsuir.RV_Project.domain.Story;
import by.bsuir.RV_Project.dto.requests.StoryRequestDto;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface StoryRequestConverter {
    @Mapping(source = "authorId", target = "author.id")
    Story fromDto(StoryRequestDto story);
}

File: 151003/Yagnnish/Lab1/Dtos/RequestDtos/RequestConverters/StoryRequestConverter.cs
using lab_1.Domain;

namespace lab_1.Dtos.RequestDtos.RequestConverters
{
    public class StoryRequestConverter : BaseRequest<Story, StoryRequestDto>
    {
        public Story FromDto(StoryRequestDto dto, long? id)
        {
            return new Story(id, dto.authorId, dto.title, dto.content,DateTime.Now, DateTime.Now);
        }
    }
}
